### ISSUE TITLE
Pizza ordering allowed by non-registered people

### ISSUE THREAD

**Issue by Komkommer123** [svthalia/concrexit#1928]
### Describe the bug
Pizza order allowed for other event that you are not registered for. If someone registers for a later event that they cannot yet order pizza for, they can order pizza for an event that does allow ordering of pizza.

### How to reproduce
Steps to reproduce the behaviour:
1. Register for event with pizza 2, where event 2 has not opened their order window
2. Order pizza for event with pizza 1, where event 1 has opened their order window
3. Pizza gets delivered to pizza event 1
4. Person is not there at event 1 but pizza is in the order list for that event.

### Expected behaviour
People not registered for an event should not be able to order pizza for said event.

### Additional context
This loses Thalia money, as pizza's get ordered that are not paid for.

**Comment by se-bastiaan** [comment]
Did this actually happen?

This sounds a bit as user error to me. The website clearly states which event you are ordering pizza for, and the orders are meant to take place during the event. If it is unclear which event has the order window open, then that means that the functionality is used in a way that it was never intended to.

Having the possibility to order pizzas even though people are not registered for an event is a valid use case and has happened numerous times in the past. Think about people that are present but a not a Thalia member, or people that are in the same room but not actively participating.

So I have no idea why this needs to be high priority, in my opinion this is even a non-issue.

**Comment by JobDoesburg** [comment]
I do see that it is bad UX that the "order a pizza" button appears on an event page while you actually cannot yet order. But it is also user-error as indeed, it is clear when ordering for what event you are ordering. 

I propose to change this issue to 'Hide pizza order button if ordering for event is not yet possible'.

Possibly, we could also implement the option to restrict orders to registered people (as a feature)

**Comment by se-bastiaan** [comment]
> I propose to change this issue to 'Hide pizza order button if ordering for event is not yet possible'.

Sounds good to me, even easy and fun. Shouldn't be so hard because it is basically checking if the FoodEvent attached to the event equals `FoodEvent.current()`

**Comment by DeD1rk** [comment]
I would say it's actually most logical if the food order page for different food events can all be opened, and the order buttons redirect to the event's food event. Then pizza.thalia.nu and thalia.nu/pizza/ should redirect to thalia.nu/pizza/<foodeventpk>/. The wier behaviour when you press the pizza button of a future event is because the buttons all redirect to a single ordering page for the current event.

But if you decide to just hide the button, I would suggest making it disabled instead with a tooltip. That way it's still clearly visible that it will be possible to order later.

**Comment by se-bastiaan** [comment]
Yes that would be logical, but we have #1654 to do it properly basically. Not sure how much work it would be to implement the food event views correctly. 

> But if you decide to just hide the button, I would suggest making it disabled instead with a tooltip. That way it's still clearly visible that it will be possible to order later.

Probably better indeed.
